iceoryx_hoofs/include/iceoryx_hoofs/cxx/optional.hpp

```diff
@@ -253,13 +243,58 @@ class optional : public FunctionalInterface<optional<T>, T, void>
     void destruct_value() noexcept;
 };
 
+// AXIVION Next Construct AutosarC++19_03-M17.0.3 : make_optional is defined within iox::cxx which prevents easy misuse
 /// @brief Creates an optional which contains a value by forwarding the arguments
 ///         to the constructor of T.
 /// @param[in] args arguments which will be perfectly forwarded to the constructor
 ///             of T
 /// @return optional which contains T constructed with args
 template <typename OptionalBaseType, typename... Targs>
 optional<OptionalBaseType> make_optional(Targs&&... args) noexcept;
+
+/// @brief Compare two optionals for equality.
+/// @param[in] lhs cxx::optional
+/// @param[in] rhs value to which this optional should be compared to
```

Comment:
```suggestion
/// @param[in] rhs The other optional to which should be compared to
```

Also for the other doxygen entries.